### PR TITLE
shortnames: correctly determine cache directory for rootless

### PR DESCRIPTION
**Pull request by go-d**
PR #1142 placed .cache as a subdirectory below configHome, but
$HOME/.cache is what it ought te be.

Signed-off-by: G.I. Doe <gogit@pm.me>

### CODE REVIEW

**Comment by go-d** [comment]
BTW, this PR depends on containers/storage#824

File: pkg/sysregistriesv2/shortnames.go [hunk 1]
```diff
@@ -34,15 +34,9 @@ func shortNameAliasesConfPath(ctx *types.SystemContext) (string, error) {
...
-		cacheRoot = xdgCache
-	} else {
-		configHome, err := homedir.GetConfigHome()
-		if err != nil {
-			return "", err
-		}
-		cacheRoot = filepath.Join(configHome, ".cache")
+	cacheRoot, err := homedir.GetCacheHome()
+	if err != nil {
+		return "", err
```
Comment (mtrmac) [inline pkg/sysregistriesv2/shortnames.go:39]: (Drive-by comment): With the current implementation, won’t that make using any of this code on Windows completely impossible? Is that OK?

Comment (vrothberg) [reply pkg/sysregistriesv2/shortnames.go:39]: At the moment yes.  Short names are always resolved on the Linux nodes.

**Review by vrothberg — COMMENTED** [review]
Can you elaborate a bit more on that?

Placing it under XDG_*CACHE*_HOME was intentional rootless users follow the XDG specification (Cc @giuseppe).

**Comment by vrothberg** [comment]
Maybe I need another cup of coffee but the PR title suggest this is fixing something.  But it seems like it's just moving the logic to c/storage. Am I missing something?

**Comment by go-d** [comment]
@vrothberg The commit message says what this fixes. Path should not be _~/.config/.cache/containers/short-name-aliases.conf.lock_ but _~/.cache/containers/short-name-aliases.conf.lock_ for rootless.

**Comment by vrothberg** [comment]
> @vrothberg The commit message says what this fixes. Path should not be _~/.config/.cache/containers/short-name-aliases.conf.lock_ but _~/.cache/containers/short-name-aliases.conf.lock_ for rootless.

Thanks! It was a lack of coffee :)  We first need to merge the c/storage PR, then we merge here with vendoring it.

**Comment by vrothberg** [comment]
I'll prepare a fix for the `release-v5.10` branch.

**Comment by vrothberg** [comment]
Can you do a `go get github.com/containers/storage@master`?  That should let the build pass and turn CI green.

File: pkg/sysregistriesv2/shortnames.go [hunk 2]
```diff
@@ -34,15 +34,9 @@ func shortNameAliasesConfPath(ctx *types.SystemContext) (string, error) {
...
-	var cacheRoot string
-	if xdgCache := os.Getenv("XDG_CACHE_HOME"); xdgCache != "" {
-		cacheRoot = xdgCache
-	} else {
-		configHome, err := homedir.GetConfigHome()
-		if err != nil {
-			return "", err
-		}
-		cacheRoot = filepath.Join(configHome, ".cache")
+	cacheRoot, err := homedir.GetCacheHome()
```
Comment (vrothberg) [inline pkg/sysregistriesv2/shortnames.go:37]: This cannot compile yet.

You need to update the dependency on `c/storage`.  Since @rhatdan cut a new release with your commit there, you can now do a `go get github.com/containers/storage@v1.26.0` and add the changes to this commit.

**Comment by go-d** [comment]
> Can you do a `go get github.com/containers/storage@master`? That should let the build pass and turn CI green.
> 
> Edit: also please rebase

I probaly did something wrong. Any pointers?

File: go.mod [hunk 3]
```diff
@@ -8,7 +8,7 @@ require (
 	github.com/BurntSushi/toml v0.3.1
 	github.com/containers/libtrust v0.0.0-20190913040956-14b96171aa3b
 	github.com/containers/ocicrypt v1.1.0
-	github.com/containers/storage v1.25.0
+	github.com/containers/storage v1.26.1-0.20210219111608-5bd1a7d801b6
```
Comment (vrothberg) [inline go.mod:11]: This should be `github.com/containers/storage v1.26.0`.

**Review by vrothberg — COMMENTED** [review]
Just one last nit, then ready to merge :)

Thanks again for tackling the bug directly!

**Review by vrothberg — COMMENTED** [review]
LGTM, thank you

**Comment by vrothberg** [comment]
@rhatdan @giuseppe PTAL

**Comment by go-d** [comment]
> Just one last nit, then ready to merge :)
> 
> Thanks again for tackling the bug directly!

Thank you for the learning experience.

**Comment by rhatdan** [comment]
LGTM
